HdProduction.HelpDesk.Api/Configuration/ErrorHandlingMiddleware.cs

```diff
@@ -20,16 +21,21 @@ protected override Task HandleExceptionAsync(HttpContext context, Exception exce
       string message;
       var statusCode = HttpStatusCode.InternalServerError;
 
-      // TODO normal filters
-      if (exception is ApplicationException)
+      switch (exception)
       {
-        statusCode = HttpStatusCode.BadRequest;
-        message = exception.Message;
-      }
-      else
-      {
-        Logger.Error("Unknown exception", exception);
-        message = "Oops. Something went wrong.";
+        // TODO normal filters
+        case EntityNotFoundException _:
+          statusCode = HttpStatusCode.NotFound;
+          message = exception.Message;
+          break;
+        case ApplicationException _:
```

Comment:
ApplicationException  was a temp thing, for bad request we usually throw BusinessLogicException (not sure that I've added it)